### PR TITLE
Upgrade to `go1.19.3`

### PR DESCRIPTION
**Pull request by frouioui**
## Description

This Pull Request upgrades the go version of Vitess from `go1.18.7` to the latest version: `go1.19.3`.

## Related Issue(s)

- Closes https://github.com/vitessio/vitess/issues/11654

## Checklist

-   [x] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by frouioui** [comment]
I have added the `do not merge` label, let's wait until we have upgraded the Golang version on arewefastyet before merging this Pull Request.

**Comment by frouioui** [comment]
Note to self: We need to open a Pull Request against `release-15.0` to change the Go version used in the upgrade/downgrade workflows. Otherwise, they will fail consistently.

**Comment by dbussink** [comment]
@frouioui Updated the generated proto with 1.19 since that was failing the static code checks.

**Review by dbussink — APPROVED** [review]
I think this is good to go once we have the benchmark results come back.

**Comment by frouioui** [comment]
@dbussink thank you!

**Comment by frouioui** [comment]
> The macrobenchmark comparison is visible here: https://benchmark.vitess.io/compare?r=86869c0805ee61628a6865a394cc56fe6bb46799&c=18c07d8f46f318550d0b28db72a9e0df3a2ce194

There is no significant different between `main` and this Pull Request. I think it is thus safe to upgrade.

File: build.env
Comment (deepthi) [inline build.env:20]: Does this _have_ to be 1.19.3+ or is any 1.19.x version acceptable?

Comment (frouioui) [reply build.env:20]: It requires `1.19.3+`, we don't have a thing to distinguish prod and dev environments to allow `1.19.*` in dev environments, yet.

**Review by deepthi — APPROVED** [review]

**Review by rohit-nayak-ps — APPROVED** [review]
lgtm

**Pull request merged**
